### PR TITLE
Custom Reports: Add filters to sessionStorage

### PR DESCRIPTION
**Pull request by carkom**
Fixing some of the session storage issues plus adding filters to sessionStorage.

### CODE REVIEW

**Comment by MatissJanis** [comment]
Would you mind giving me an example how to test this change? I tried comparing it with `master` branch, but I can't spot any difference in functionality.

**Comment by carkom** [comment]
> Would you mind giving me an example how to test this change? I tried comparing it with `master` branch, but I can't spot any difference in functionality.

Start with a new default custom report. Add a filter. Click a bar to be sent to "/accounts" and see the transactions. Go back on your browser.

In master your filter will be gone. This PR retains the filter.

**Review by MatissJanis — COMMENTED** [review]
Overall LGTM. Just a small suggestion to reduce duplication.

File: packages/desktop-client/src/components/reports/ReportSidebar.jsx [hunk 1]
```diff
@@ -199,6 +199,14 @@ export function ReportSidebar({
...
             value={customReportItems.interval}
             onChange={e => {
+              const storedReport = JSON.parse(sessionStorage.getItem('report'));
+              sessionStorage.setItem(
+                'report',
+                JSON.stringify({
+                  ...storedReport,
+                  interval: e,
+                }),
+              );
```
[the remark this answers is not in the document]
Comment (carkom) [reply packages/desktop-client/src/components/reports/ReportSidebar.jsx]: Yea, good call.

**Review by MatissJanis — APPROVED** [review]

**Pull request merged**